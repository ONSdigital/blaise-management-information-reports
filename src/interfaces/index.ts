interface InterviewerCallHistoryReportData {
    questionnaire_name: string
    serial_number: string
    call_start_time: string
    dial_secs: string
    call_result: string
}

interface AppointmentResourcePlanningReportData {
    questionnaire_name: string
    appointment_time: string
    appointment_language: string
    total: number
}

export type {InterviewerCallHistoryReportData, AppointmentResourcePlanningReportData};
