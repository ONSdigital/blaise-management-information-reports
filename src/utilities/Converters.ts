// eslint-disable-next-line import/prefer-default-export
export function convertSecondsToMinutesAndSeconds(seconds: number): string {
    const convertedMinutes = (`0${Math.floor(seconds / 60)}`).slice(-2);
    const convertedSeconds = (`0${seconds - (Math.floor(seconds / 60) * 60)}`).slice(-2);
    return `${convertedMinutes}:${convertedSeconds}`;
}
