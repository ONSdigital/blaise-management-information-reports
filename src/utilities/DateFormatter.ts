import dayjs from "dayjs";
import dateFormatter from "dayjs";
import utc from "dayjs/plugin/utc";
import timezone from "dayjs/plugin/timezone";
import customParseFormat from "dayjs/plugin/customParseFormat";

dateFormatter.extend(utc);
dateFormatter.extend(timezone);
dateFormatter.extend(customParseFormat);

export default function subtractYears(numOfYears: number, date: Date = new Date()): Date {
    return dayjs(date).subtract(numOfYears, "year").toDate();
}

export function formatDate(date: Date | string): string {
    return dateFormatter(date).tz("Europe/London").format("DD/MM/YYYY");
}

export function formatDateAndTime(date: Date | string): string {
    return dateFormatter(date).tz("Europe/London").format("DD/MM/YYYY HH:mm:ss");
}

export function formatISODate(date: Date | string): string {
    return dateFormatter(date).tz("Europe/London").format("YYYY-MM-DD");
}
