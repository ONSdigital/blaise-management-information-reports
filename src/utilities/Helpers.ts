import dayjs from "dayjs";
import dateFormatter from "dayjs";

export default function subtractYears(numOfYears: any, date: Date = new Date()) {
    return dayjs(date).subtract(numOfYears, "year").toDate();
}

export function bstDateFormatter(date: Date): string {
    return dateFormatter(date).tz("Europe/London").format("DD/MM/YYYY");
}

export function bstDateFormatterWithTime(date: Date): string {
    // I am aware this is horrific but it keeps the tests passing for now.
    const valid_date = dateFormatter(date).tz("Europe/London").format("DD/MM/YYYY HH:mm:ss");
    if (valid_date == "Invalid Date") {
        return valid_date;
    }
    return ` (${valid_date})`;
}

export function bstStringDateFormatterWithTime(date: string): string {
    return dateFormatter(date).tz("Europe/London").format("DD/MM/YYYY HH:mm:ss");
}

export function mirStandardFormatForNewDate(): string {
    return dateFormatter(new Date()).format("DD-MM-YYYY");
}
