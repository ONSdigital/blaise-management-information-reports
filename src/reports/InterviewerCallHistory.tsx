import React, {ReactElement, useState} from "react";
import {Link} from "react-router-dom";
import {ONSButton, ONSPanel} from "blaise-design-system-react-components";
import FormTextInput from "../form/TextInput";
import Form from "../form";
import {requiredValidator} from "../form/FormValidators";
import {getReport} from "../utilities/http";
import {ErrorBoundary} from "../Components/ErrorHandling/ErrorBoundary";
import {ONSDateInput} from "../Components/ONSDesignSystem/ONSDateInput";

function InterviewerCallHistory(): ReactElement {
    const [buttonLoading, setButtonLoading] = useState<boolean>(false);
    const [interviewerID, setInterviewerID] = useState<string>("");
    const [startDate, setStartDate] = useState<Date>(new Date());
    const [endDate, setEndDate] = useState<Date>(new Date());
    const [message, setMessage] = useState<string>("");
    const [listReportData, setListReportData] = useState<any[]>([]);

    async function runReport(formData: any) {
        console.log(formData);
        setInterviewerID(formData.interviewer);
        formData.start_date = startDate;
        formData.end_date = endDate;

        console.log(`runReport ${formData}`);

        const [success, list] = await getReport(formData);

        if (!success) {
            setMessage("Error running report");
            return;
        }

        console.log(list);
        setListReportData(list);

    }

    return (
        <>
            <p className="u-mt-m">
                <Link to={"/"}>Previous</Link>
            </p>
            <h1>Run interviewer call history report</h1>
            <ONSPanel hidden={(message === "")} status="error">
                {message}
            </ONSPanel>
            <Form onSubmit={(data) => runReport(data)}>
                <p>
                    <FormTextInput
                        name="interviewer"
                        validators={[requiredValidator]}
                        label={"Interviewer ID"}
                    />
                </p>
                <ONSDateInput
                    label={"Start Date"}
                    date={startDate}
                    onChange={(date) => setStartDate(date)}
                />
                <br/>
                <ONSDateInput
                    label={"End Date"}
                    date={endDate}
                    onChange={(date) => setEndDate(date)}
                />
                <br/>
                <br/>
                <ONSButton
                    label={"Run"}
                    primary={true}
                    loading={buttonLoading}
                    submit={true}/>
            </Form>
            <br/>

            <ErrorBoundary errorMessageText={"Failed to load"}>
                {
                    listReportData && listReportData.length > 0
                        ?
                        <table id="batches-table" className="table ">
                            <thead className="table__head u-mt-m">
                            <tr className="table__row">
                                {/*
                                <th scope="col" className="table__header ">
                                    <span>Interviewer ID</span>
                                </th>
                                */}
                                <th scope="col" className="table__header ">
                                    <span>Questionnaire</span>
                                </th>
                                <th scope="col" className="table__header ">
                                    <span>Serial Number</span>
                                </th>
                                <th scope="col" className="table__header ">
                                    <span>Call Start Time</span>
                                </th>
                                <th scope="col" className="table__header ">
                                    <span>Call Length</span>
                                </th>
                                <th scope="col" className="table__header ">
                                    <span>Interviews</span>
                                </th>
                                <th scope="col" className="table__header ">
                                    <span>Call Result</span>
                                </th>
                            </tr>
                            </thead>
                            <tbody className="table__body">
                            {
                                listReportData.map((batch: any) => {
                                    return (
                                        <tr className="table__row" key={batch.name}
                                            data-testid={"batches-table-row"}>
                                            {/*
                                            <td className="table__cell ">
                                                {batch.interviewer}
                                            </td>
                                            */}
                                            <td className="table__cell ">
                                                {batch.questionnaire_name}
                                            </td>
                                            <td className="table__cell ">
                                                {batch.serial_number}
                                            </td>
                                            <td className="table__cell ">
                                                {batch.call_start_time}
                                            </td>
                                            <td className="table__cell ">
                                                {batch.dial_secs}
                                            </td>
                                            <td className="table__cell ">
                                                {batch.number_of_interviews}
                                            </td>
                                            <td className="table__cell ">
                                                {batch.call_result}
                                            </td>
                                        </tr>
                                    );
                                })
                            }
                            </tbody>
                        </table>
                        :
                        <br/>
                }
            </ErrorBoundary>

        </>
    );
}

export default InterviewerCallHistory;
