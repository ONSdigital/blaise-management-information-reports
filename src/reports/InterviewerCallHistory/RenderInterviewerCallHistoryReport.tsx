import React, { ReactElement, useEffect, useState } from "react";
import Breadcrumbs from "../../components/Breadcrumbs";
import { CSVLink } from "react-csv";
import { ErrorBoundary, ONSPanel } from "blaise-design-system-react-components";
import { InterviewerCallHistoryReport } from "../../interfaces";
import { convertSecondsToMinutesAndSeconds } from "../../utilities/Converters";
import { getInterviewerCallHistoryReport } from "../../utilities/HTTP";
import CallHistoryLastUpdatedStatus from "../../components/CallHistoryLastUpdatedStatus";
import { formatDate, formatDateAndTime } from "../../utilities/DateFormatter";

interface RenderInterviewerCallHistoryReportPageProps {
    interviewer: string
    startDate: Date
    endDate: Date
    surveyTla: string
    questionnaires: string[]
    navigateBack: () => void
    navigateBackTwoSteps: () => void
}

function formatList(listOfQuestionnaires: string[]): string {
    if (listOfQuestionnaires.length === 1) return listOfQuestionnaires[0];
    const firsts = listOfQuestionnaires.slice(0, listOfQuestionnaires.length - 1);
    const last = listOfQuestionnaires[listOfQuestionnaires.length - 1];
    return firsts.join(", ") + " and " + last;
}

function RenderInterviewerCallHistoryReport(props: RenderInterviewerCallHistoryReportPageProps): ReactElement {
    const [reportData, setReportData] = useState<InterviewerCallHistoryReport[]>([]);
    const [interviewerID, setInterviewerID] = useState<string>("");
    const [messageNoData, setMessageNoData] = useState<string>("");
    const {
        interviewer,
        startDate,
        endDate,
        questionnaires,
        navigateBack,
        navigateBackTwoSteps,
    } = props;

    const reportExportHeaders = [
        { label: "Interviewer", key: "interviewer" },
        { label: "Questionnaire", key: "questionnaire_name" },
        { label: "Serial Number", key: "serial_number" },
        { label: "Call Start Time", key: "call_start_time" },
        { label: "Call Length (Seconds)", key: "dial_secs" },
        { label: "Call Result", key: "call_result" }
    ];

    useEffect(() => {
        runInterviewerCallHistoryReport();
    }, []
    );

    async function runInterviewerCallHistoryReport(): Promise<void> {
        const formValues: Record<string, any> = {};
        setMessageNoData("");
        setReportData([]);
        setInterviewerID(props.interviewer);
        formValues.survey_tla = props.surveyTla;
        formValues.interviewer = props.interviewer;
        formValues.start_date = props.startDate;
        formValues.end_date = props.endDate;
        formValues.questionnaires = props.questionnaires;

        let callHistory: InterviewerCallHistoryReport[];
        try {
            callHistory = await getInterviewerCallHistoryReport(formValues);
        } catch {
            //setReportFailed(true);
            return;
        } finally {
            //setSubmitting(false);
        }

        if (callHistory.length === 0) {
            setMessageNoData("No data found for parameters given.");
            return;
        }

        console.log(callHistory);
        setReportData(callHistory);

    }

    return (
        <>
            <Breadcrumbs BreadcrumbList={[{ link: "/", title: "Reports" }, {
                link: "#",
                onClickFunction: navigateBackTwoSteps,
                title: "Interviewer details"
            }, { link: "#", onClickFunction: navigateBack, title: "Questionnaires" }]}/>
            <main id="main-content" className="page__main u-mt-s">
                <h1>Call History Report</h1>
                <h3 className="u-mb-m">
                    Interviewer: {interviewer} <br></br>
                    Period: {formatDate(startDate)}â€“{formatDate(endDate)}<br></br>
                    Questionnaire{questionnaires.length > 1 ? ("s") : ""}: {formatList(questionnaires)}
                </h3>
                <CallHistoryLastUpdatedStatus/>

                <div>

                </div>

                <br/>
                <CSVLink hidden={reportData === null || reportData.length === 0}
                    data={
                        reportData?.map(row => (
                            { ...row, call_start_time: formatDateAndTime(row.call_start_time) }
                        ))
                    }
                    headers={reportExportHeaders}
                    target="_blank"
                    filename={`interviewer-call-history-${interviewerID}.csv`}>
                    Export report as Comma-Separated Values (CSV) file
                </CSVLink>

                <ErrorBoundary errorMessageText={"Failed to load"}>
                    {
                        reportData && reportData.length > 0
                            ?
                            <table id="report-table" className="table u-mt-s">
                                <thead className="table__head u-mt-m">
                                    <tr className="table__row">
                                        <th scope="col" className="table__header ">
                                            <span>Questionnaire</span>
                                        </th>
                                        <th scope="col" className="table__header ">
                                            <span>Serial Number</span>
                                        </th>
                                        <th scope="col" className="table__header ">
                                            <span>Call Start Time</span>
                                        </th>
                                        <th scope="col" className="table__header ">
                                            <span>Call Length</span>
                                        </th>
                                        <th scope="col" className="table__header ">
                                            <span>Call Result</span>
                                        </th>
                                    </tr>
                                </thead>
                                <tbody className="table__body">
                                    {
                                        reportData.map((callHistory: InterviewerCallHistoryReport) => {
                                            return (
                                                <tr className="table__row" key={callHistory.call_start_time}
                                                    data-testid={"report-table-row"}>
                                                    <td className="table__cell ">
                                                        {callHistory.questionnaire_name}
                                                    </td>
                                                    <td className="table__cell ">
                                                        {callHistory.serial_number}
                                                    </td>
                                                    <td className="table__cell ">
                                                        {formatDateAndTime(callHistory.call_start_time)}
                                                    </td>
                                                    <td className="table__cell ">
                                                        {convertSecondsToMinutesAndSeconds(callHistory.dial_secs)}
                                                    </td>
                                                    <td className="table__cell ">
                                                        {(callHistory.call_result === null ? "Unknown" : callHistory.call_result)}
                                                    </td>
                                                </tr>
                                            );
                                        })
                                    }
                                </tbody>
                            </table>
                            :
                            <ONSPanel hidden={messageNoData === "" && true}>{messageNoData}</ONSPanel>
                    }
                    <br/>
                </ErrorBoundary>
            </main>
        </>
    );
}

export default RenderInterviewerCallHistoryReport;
