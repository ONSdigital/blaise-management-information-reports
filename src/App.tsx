import React, { ReactElement, useEffect, useState } from "react";
import { Link, Route, Switch, useLocation } from "react-router-dom";
import { BetaBanner, DefaultErrorBoundary, Footer, Header } from "blaise-design-system-react-components";
import InterviewerCallHistory from "./reports/InterviewerCallHistory";
import InterviewerCallPattern from "./reports/InterviewerCallPattern";
import AppointmentResourcePlanning from "./reports/AppointmentResourcePlanning/AppointmentResourcePlanning";
import "./style.css";
import LoginForm from "./components/LoginForm";
import { useToken, clearToken } from "./client/token";
import { validateToken } from "./client/user";

const divStyle = {
    minHeight: "calc(72vh)"
};

function App(): ReactElement {
    const location = useLocation();
    const { token, setToken } = useToken();
    const [loggedIn, setLoggedIn] = useState(false);

    useEffect(() => {
        console.log(location);
        validateToken(token).then((validated: boolean) => {
            setLoggedIn(validated);
        }).catch((error: unknown) => {
            console.log(`Error checking logged in state: ${error}`);
            setLoggedIn(false);
        });
    });

    function loginPage(): ReactElement {
        if (loggedIn) {
            return <></>;
        }
        return <LoginForm setToken={setToken} />;
    }

    function signOut(): void {
        clearToken();
        setToken(null);
        setLoggedIn(false);
    }

    function app(): ReactElement | undefined {
        if (loggedIn) {
            return (<DefaultErrorBoundary>
                <Switch>
                    <Route path="/interviewer-call-history">
                        <InterviewerCallHistory />
                    </Route>
                    <Route path="/interviewer-call-pattern">
                        <InterviewerCallPattern />
                    </Route>
                    <Route path="/appointment-resource-planning">
                        <AppointmentResourcePlanning />
                    </Route>
                    <Route path="/">
                        <main id="main-content" className="page__main u-mt-no">
                            <h1 className="u-mt-m">Reports</h1>
                            <div className="grid grid--column@xxs@s u-mt-m">
                                <div className="grid__col col-6@m">
                                    <div className="card" aria-labelledby="interviewer-call-history"
                                        aria-describedby="interviewer-call-history-text">
                                        <h2 className="u-fs-m" id="interviewer-call-history">
                                            <Link to="/interviewer-call-history">
                                                Interviewer call history
                                            </Link>
                                        </h2>
                                        <p id="interviewer-call-history-text">Generate report to see an interviewers
                                            call history over a given date range.</p>
                                    </div>
                                </div>
                                <div className="grid__col col-6@m">
                                    <div className="card" aria-labelledby="interviewer-call-pattern"
                                        aria-describedby="interviewer-call-pattern-text">
                                        <h2 className="u-fs-m" id="interviewer-call-pattern">
                                            <Link to="/interviewer-call-pattern">
                                                Interviewer call pattern
                                            </Link>
                                        </h2>
                                        <p id="interviewer-call-pattern-text">Generate report to analyse
                                            productivity of an interviewer over a given date range.</p>
                                    </div>
                                </div>
                            </div>
                            <div className="grid grid--column@xxs@s  u-mt-m">
                                <div className="grid__col col-6@m">
                                    <div className="card" aria-labelledby="appointment-resource-planning"
                                        aria-describedby="appointment-resource-planning-text">
                                        <h2 className="u-fs-m" id="appointment-resource-planning">
                                            <Link to="/appointment-resource-planning">
                                                Appointment resource planning
                                            </Link>
                                        </h2>
                                        <p id="appointment-resource-planning-text">Generate report to view the
                                            number of interview appointments scheduled for a given date.</p>
                                    </div>
                                </div>
                            </div>
                        </main>
                    </Route>
                </Switch>
            </DefaultErrorBoundary>
            );
        }
        return undefined;
    }

    return (
        <>
            <a className="skip__link" href="#main-content">Skip to main content</a>
            <BetaBanner />
            <Header title={"Management Information Reports"} signOutButton={true} noSave={true} signOutFunction={signOut} />
            <div style={divStyle} className="page__container container">
                {loginPage()}
                {app()}
            </div>
            <Footer />
        </>
    );
}

export default App;
