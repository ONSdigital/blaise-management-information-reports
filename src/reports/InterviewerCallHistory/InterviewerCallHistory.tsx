import React, { ReactElement, useState } from "react";
import InterviewerFilter, { InterviewerFilterQuery } from "../filters/InterviewerFilter";
import QuestionnaireFilter from "../filters/QuestionnaireFilter";
import RenderInterviewerCallHistoryReport from "./RenderInterviewerCallHistoryReport";

enum Step {
    InterviewerFilter,
    QuestionnaireFilter,
    RenderReport,
}

function InterviewerCallHistory(): ReactElement {
    const [activeStep, setActiveStep] = useState<Step>(Step.InterviewerFilter);
    const [interviewerFilterQuery, setInterviewerFilterQuery] = useState<InterviewerFilterQuery>({
        interviewer: "",
        startDate: new Date(),
        endDate: new Date(),
        surveyTla: "",
    });
    const [questionnaires, setQuestionnaires] = useState<string[]>([]);

    function _renderStepContent(step: number) {
        switch (step) {
        case Step.InterviewerFilter:
            return (
                <InterviewerFilter
                    title="call history"
                    query={interviewerFilterQuery}
                    onSubmit={_handleInterviewerFilterSubmit}
                />
            );
        case Step.QuestionnaireFilter:
            return (
                <QuestionnaireFilter
                    interviewerFilterQuery={interviewerFilterQuery}
                    questionnaires={questionnaires}
                    setQuestionnaires={setQuestionnaires}
                    onSubmit={_handleQuestionnaireFilterSubmit}
                    navigateBack={_navigateBack}
                />
            );
        case Step.RenderReport:
            console.log(`Steps questionnaires ${questionnaires}`);
            return (
                <RenderInterviewerCallHistoryReport
                    interviewerFilterQuery={interviewerFilterQuery}
                    questionnaires={questionnaires}
                    navigateBack={_navigateBack}
                    navigateBackTwoSteps={_navigateBackTwoSteps}
                />
            );
        }
    }

    function _handleInterviewerFilterSubmit(query: InterviewerFilterQuery) {
        setInterviewerFilterQuery(query);
        setActiveStep(Step.QuestionnaireFilter);
    }

    function _handleQuestionnaireFilterSubmit() {
        setActiveStep(Step.RenderReport);
    }

    function _navigateBack() {
        setActiveStep(activeStep - 1);
    }

    function _navigateBackTwoSteps() {
        setActiveStep(activeStep - 2);
    }

    return (
        <div>
            <div className="u-mt-m">
                { _renderStepContent(activeStep) }
            </div>
        </div>
    );
}

export default InterviewerCallHistory;
