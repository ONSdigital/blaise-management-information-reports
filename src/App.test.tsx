import React from "react";
import {cleanup, render, waitFor} from "@testing-library/react";
import App from "./App";
import "@testing-library/jest-dom";
import flushPromises from "./tests/utils";
import {act} from "react-dom/test-utils";
import {createMemoryHistory} from "history";
import {Router} from "react-router";
import {convertSecondsToMinutesAndSeconds} from "./utilities/http";
import InterviewerCallHistory from "./reports/InterviewerCallHistory";
import {ReportData} from "./interfaces";

test("test 60 seconds is converted to 01:00", () => {
    expect(convertSecondsToMinutesAndSeconds(60)).toBe("01:00");
});

test("test 61 seconds is converted to 01:01", () => {
    expect(convertSecondsToMinutesAndSeconds(61)).toBe("01:01");
});

test("test 120 seconds is converted to 02:00", () => {
    expect(convertSecondsToMinutesAndSeconds(120)).toBe("02:00");
});

test("test 121 seconds is converted to 02:01", () => {
    expect(convertSecondsToMinutesAndSeconds(121)).toBe("02:01");
});

test("test 3600 seconds is converted to 60:00", () => {
    expect(convertSecondsToMinutesAndSeconds(3600)).toBe("60:00");
});

test("test 3601 seconds is converted to 60:01", () => {
    expect(convertSecondsToMinutesAndSeconds(3601)).toBe("60:01");
});

test("test 0 seconds is converted to 00:00", () => {
    expect(convertSecondsToMinutesAndSeconds(0)).toBe("00:00");
});

test("test empty string is converted to 00:00", () => {
    expect(convertSecondsToMinutesAndSeconds("")).toBe("00:00");
});

describe("management information reports homepage", () => {

    it("matches snapshot", async () => {
        const history = createMemoryHistory();
        const wrapper = render(
            <Router history={history}>
                <App/>
            </Router>
        );

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    it("renders correctly", async () => {
        const history = createMemoryHistory();
        const {getByText, queryByText} = render(
            <Router history={history}>
                <App/>
            </Router>
        );

        expect(queryByText(/Management Information Reports/)).toBeInTheDocument();
        expect(queryByText(/Interviewer Call History/)).toBeInTheDocument();

    });

    afterAll(() => {
        cleanup();
    });
});

