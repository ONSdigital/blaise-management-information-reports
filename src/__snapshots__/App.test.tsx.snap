// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`management information reports homepage matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="phase-banner"
      >
        <div
          class="container "
        >
          <div
            class="grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap"
          >
            <div
              class="grid__col col-auto u-flex-no-grow"
            >
              <h3
                class="phase-banner__badge"
              >
                BETA
              </h3>
            </div>
            <div
              class="grid__col col-auto u-flex-shrink"
            >
              <p
                class="phase-banner__desc u-fs-s u-mb-no"
              >
                This is a new service – your 
                <a
                  href="https://ons.service-now.com/"
                >
                  feedback
                </a>
                 will help us improve it.
              </p>
            </div>
          </div>
        </div>
      </div>
      <header
        class="header header--internal"
      >
        <div
          class="header__top"
          role="banner"
        >
          <div
            class="container"
          >
            <div
              class="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap "
            >
              <div
                class="grid__col col-auto"
              >
                <a
                  class="header__logo-link"
                  href="/"
                >
                  <picture>
                    <img
                      alt="Office for National Statistics logo"
                      class="header__logo"
                      src="https://cdn.ons.gov.uk/sdc/design-system/31.4.0/img/ons-logo-neg-en.svg"
                    />
                  </picture>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="header__main"
        >
          <div
            class="container"
          >
            <div
              class="grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap"
            >
              <div
                class="grid__col col-auto u-flex-shrink"
              >
                <div
                  class="header__title"
                >
                  Management Information Reports
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
      <div
        class="page__container container"
        style="min-height: calc(67vh);"
      >
        <main
          class="page__main"
          id="main-content"
        >
          <div
            class="panel panel--info panel--no-title  u-mt-m"
          >
            <div
              class="panel__body "
            >
              <p>
                Data in these reports could be up to 24 hours out of date.
                <br />
                <br />
                These reports only go back to the last 12 months.
              </p>
            </div>
          </div>
          <br />
          <div
            class="grid grid--column@xxs@s u-mt-m"
          >
            <div
              class="grid__col col-6@m"
            >
              <div
                aria-describedby="interviewer-call-history-text"
                aria-labelledby="interviewer-call-history"
                class="card"
              >
                <h2
                  class="u-fs-m"
                  id="interviewer-call-history"
                >
                  <a
                    href="/interviewer-call-history"
                  >
                    Interviewer Call History
                  </a>
                </h2>
                <p
                  id="interviewer-call-history-text"
                >
                  Generate report to see an interviewers call history over a given date range.
                </p>
              </div>
            </div>
            <div
              class="grid__col col-6@m"
            >
              <div
                aria-describedby="interviewer-call-pattern-text"
                aria-labelledby="interviewer-call-pattern"
                class="card"
              >
                <h2
                  class="u-fs-m"
                  id="interviewer-call-pattern"
                >
                  <a
                    href="/interviewer-call-pattern"
                  >
                    Interviewer Call Pattern
                  </a>
                </h2>
                <p
                  id="interviewer-call-pattern-text"
                >
                  Generate report to analyse productivity of an interviewer over a given date range.
                </p>
              </div>
            </div>
          </div>
          <div
            class="grid grid--column@xxs@s  u-mt-m"
          />
        </main>
      </div>
      <footer
        class="footer"
      >
        <div
          class="footer__body"
          data-analytics="footer"
        >
          <div
            class="container "
          >
            <div
              class="grid grid--flex grid--between"
            >
              <div
                class="grid__col"
              >
                <div
                  class="u-mt-m u-mb-s"
                >
                  <img
                    alt="Office for National Statistics"
                    src="https://ons-design-system.netlify.app/img/ons-logo-black-en.svg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="phase-banner"
    >
      <div
        class="container "
      >
        <div
          class="grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap"
        >
          <div
            class="grid__col col-auto u-flex-no-grow"
          >
            <h3
              class="phase-banner__badge"
            >
              BETA
            </h3>
          </div>
          <div
            class="grid__col col-auto u-flex-shrink"
          >
            <p
              class="phase-banner__desc u-fs-s u-mb-no"
            >
              This is a new service – your 
              <a
                href="https://ons.service-now.com/"
              >
                feedback
              </a>
               will help us improve it.
            </p>
          </div>
        </div>
      </div>
    </div>
    <header
      class="header header--internal"
    >
      <div
        class="header__top"
        role="banner"
      >
        <div
          class="container"
        >
          <div
            class="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap "
          >
            <div
              class="grid__col col-auto"
            >
              <a
                class="header__logo-link"
                href="/"
              >
                <picture>
                  <img
                    alt="Office for National Statistics logo"
                    class="header__logo"
                    src="https://cdn.ons.gov.uk/sdc/design-system/31.4.0/img/ons-logo-neg-en.svg"
                  />
                </picture>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="header__main"
      >
        <div
          class="container"
        >
          <div
            class="grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap"
          >
            <div
              class="grid__col col-auto u-flex-shrink"
            >
              <div
                class="header__title"
              >
                Management Information Reports
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
    <div
      class="page__container container"
      style="min-height: calc(67vh);"
    >
      <main
        class="page__main"
        id="main-content"
      >
        <div
          class="panel panel--info panel--no-title  u-mt-m"
        >
          <div
            class="panel__body "
          >
            <p>
              Data in these reports could be up to 24 hours out of date.
              <br />
              <br />
              These reports only go back to the last 12 months.
            </p>
          </div>
        </div>
        <br />
        <div
          class="grid grid--column@xxs@s u-mt-m"
        >
          <div
            class="grid__col col-6@m"
          >
            <div
              aria-describedby="interviewer-call-history-text"
              aria-labelledby="interviewer-call-history"
              class="card"
            >
              <h2
                class="u-fs-m"
                id="interviewer-call-history"
              >
                <a
                  href="/interviewer-call-history"
                >
                  Interviewer Call History
                </a>
              </h2>
              <p
                id="interviewer-call-history-text"
              >
                Generate report to see an interviewers call history over a given date range.
              </p>
            </div>
          </div>
          <div
            class="grid__col col-6@m"
          >
            <div
              aria-describedby="interviewer-call-pattern-text"
              aria-labelledby="interviewer-call-pattern"
              class="card"
            >
              <h2
                class="u-fs-m"
                id="interviewer-call-pattern"
              >
                <a
                  href="/interviewer-call-pattern"
                >
                  Interviewer Call Pattern
                </a>
              </h2>
              <p
                id="interviewer-call-pattern-text"
              >
                Generate report to analyse productivity of an interviewer over a given date range.
              </p>
            </div>
          </div>
        </div>
        <div
          class="grid grid--column@xxs@s  u-mt-m"
        />
      </main>
    </div>
    <footer
      class="footer"
    >
      <div
        class="footer__body"
        data-analytics="footer"
      >
        <div
          class="container "
        >
          <div
            class="grid grid--flex grid--between"
          >
            <div
              class="grid__col"
            >
              <div
                class="u-mt-m u-mb-s"
              >
                <img
                  alt="Office for National Statistics"
                  src="https://ons-design-system.netlify.app/img/ons-logo-black-en.svg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
